lib/src/native/realm_core.dart

```diff
@@ -205,12 +167,9 @@ class _RealmCore {
     _realmLib.realm_scheduler_perform_work(schedulerHandle._pointer);
   }
 
-  void setScheduler(Configuration config, SchedulerHandle schedulerHandle) {
-    _realmLib.realm_config_set_scheduler(config.handle._pointer, schedulerHandle._pointer);
-  }
-
-  RealmHandle openRealm(Configuration config) {
-    final realmPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_open(config.handle._pointer), "Error opening realm at path ${config.path}");
+  RealmHandle openRealm(Configuration config, Scheduler scheduler) {
+    final configHandle = _createConfig(config, scheduler.handle);
```

Comment:
I agree. And also, creating a new `Config` each time we open a `Realm `would solve the issue with shared scheduler and we wont need anymore `config.isInUse` we wont need to throw `RealmStateError("A Realm instance for this configuraiton object already exists.");` But keep in mind that all the members in Configuration have to be `final` since setting them later will cause no update in the native config.

Reply:
Also, can we also move the scheduler under the configuration class? In case we have only one realm opened per config it wont be a problem. On this way we will be consistent with C-API where the scheduler is part of the configuration.

Reply:
lets keep the scheduler in `realm_class`. now configuration is just a data bag.